tests/Feature/MovieTest.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Tests\Feature;
+
+use Tests\TestCase;
+use App\Models\User;
+use App\Models\Movie;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+
+class MovieTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /** @test */
+    public function an_admin_can_see_movies()
+    {
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->get('/movies')
+            ->assertOk();
+    }
+
+    /** @test */
+    public function an_admin_can_create_a_movie()
+    {
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->get('/movies/create')
+            ->assertOk();
+    }
+
+    /** @test */
+    public function a_non_admin_cannot_create_a_movie()
+    {
+        $user = User::factory()->create([
+            'is_admin' => false
+        ]);
+
+        $this->actingAs($user)
+            ->get('/movies/create')
+            ->assertStatus(403);
+    }
+
+    /** @test */
+    public function an_admin_can_add_a_movie()
+    {
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->postJson('/movies', [
+                'title' => 'Sample Movie',
+                'synopsis' => 'This is a Sample Synopsis',
+                'year' => 2021,
+                'poster' => 'https://via.placeholder.com/600x750.png/00aa33?text=totam',
+                'trailer' => 'http://www.goyette.biz/',
+                'duration' => '190',
+            ])->assertOk();
+
+        $this->assertDatabaseHas('movies', [
+            'title' => 'Sample Movie',
+            'synopsis' => 'This is a Sample Synopsis',
+        ]);
+    }
+
+    /** @test */
+    public function an_admin_can_edit_a_movie()
+    {
+        $movie = Movie::factory()->create();
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->get("/movies/{$movie->id}/edit")
+            ->assertOk();
+    }
+
+    /** @test */
+    public function a_non_admin_cannot_edit_a_movie()
```

Comment:
This as well, maybe you could do something like a_user_is_not_authorised_to_see_the_edit_view